cmd/k3d/create.go

```diff
@@ -1034,12 +1035,25 @@ func runK3d(cmd *cobra.Command, args []string) error {
 		log.Info().Msg("applying the registry application to argocd")
 		registryApplicationObject := argocd.GetArgoCDApplicationObject(gitopsRepoURL, fmt.Sprintf("registry/%s", clusterNameFlag))
 
-		_, err = argocdClient.ArgoprojV1alpha1().Applications("argocd").Create(context.Background(), registryApplicationObject, metav1.CreateOptions{})
+		err = k3d.RestartDeployment(context.Background(), kcfg.Clientset, "argocd", "argocd-applicationset-controller")
+		if err != nil {
+			return fmt.Errorf("error in restarting argocd controller %w", err)
+		}
+
+		err = wait.PollImmediate(5*time.Second, 20*time.Second, func() (bool, error) {
+			_, err := argocdClient.ArgoprojV1alpha1().Applications("argocd").Create(context.Background(), registryApplicationObject, metav1.CreateOptions{})
+			if err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
```

Comment:
What's the expectation here? If the application couldn't be created, try again?

If so, can we scope down the rules a bit? If it couldn't be created cause it already exists, for example, would have no point on keep trying, right?

If you can give some more context I can scope it down.